src/inference/ns.jl

```diff
@@ -0,0 +1,354 @@
+    logL, idx = findmin(spl.active_logl)
+    draw = spl.active_points[:, idx]
+    log_wt = spl.log_vol + logL
+
+    # update sampler
+    logz = logaddexp(spl.logz, log_wt)
+    spl.h = (exp(log_wt - logz) * logL +
+           exp(spl.logz - logz) * (spl.h + spl.logz) - logz)
+    spl.logz = logz
+
+    return NestedTransition(draw, logL, log_wt)     ## or `Transition(spl)`
+end
+
+function AbstractMCMC.step!(    ## check 2 AbstractMCMC.step! functions ?
+    rng::AbstractRNG,    
+    model::Model,    ## or `NestedModel`
+    spl::Sampler{<:NS},      ## or `Nested{T,B}`
+    ::Integer,
+    prev::NestedTransition;    ## or `prev::Transition` or only `transition`
+    iteration,
+    debug::Bool = false,
+    kwargs...
+) where {T, P, B}
+
+    # Find least likely point
+    logL, idx = findmin(spl.active_logl)
+    draw = spl.active_points[:, idx]
+    log_wt = spl.log_vol + logL
+
+    # update evidence and information
+    logz = logaddexp(spl.logz, prev.log_wt)
+    spl.h = (exp(prev.log_wt - logz) * prev.logL +
+           exp(spl.logz - logz) * (spl.h + spl.logz) - logz)
+    spl.logz = logz
+
+    # check if ready for first update
+    if !spl.has_bounds && spl.ncall > spl.min_ncall && iteration / spl.ncall < spl.min_eff
+        debug && @info "First update: it=$iteration, ncall=$(spl.ncall), eff=$(iteration / spl.ncall)"
+        spl.has_bounds = true
+        pointvol = exp(spl.log_vol) / spl.nactive
+        spl.active_bound = NestedSamplers.Bounds.scale!(NestedSamplers.Bounds.fit(B, spl.active_us, pointvol = pointvol), spl.enlarge)   ## `NestedSamplers.Bounds` or only `Bounds`
+        spl.since_update = 0
+    # if accepted first update, is it time to update again?
+    elseif iszero(spl.since_update % spl.update_interval)
+        debug && @info "Updating bounds: it=$iteration, ncall=$(spl.ncall), eff=$(iteration / spl.ncall)"
+        pointvol = exp(spl.log_vol) / spl.nactive
+        spl.active_bound = NestedSamplers.Bounds.scale!(NestedSamplers.Bounds.fit(B, spl.active_us, pointvol = pointvol), spl.enlarge)
+        spl.since_update = 0
+    end
+    
+    # Get a live point to use for evolving with proposal
+    if spl.has_bounds
+        point, bound = rand_live(rng, spl.active_bound, spl.active_us)
+        u, v, logl, ncall = spl.proposal(rng, point, logL, bound, model.loglike, model.prior_transform)
+    else
+        point = rand(rng, T, s.ndims)
+        bound = NestedSamplers.Bounds.NoBounds(T, spl.ndims)
+        proposal = NestedSamplers.Proposals.Uniform()
+        u, v, logl, ncall = proposal(rng, point, logL, bound, model.loglike, model.prior_transform)
+    end
+
+    # Get new point and log like
+    spl.active_us[:, idx] = u
+    spl.active_points[:, idx] = v
+    spl.active_logl[idx] = logl
+    spl.ndecl = log_wt < prev.log_wt ? spl.ndecl + 1 : 0
+    spl.ncall += ncall
+    spl.since_update += 1
+
+    # Shrink interval
+    spl.log_vol -=  1 / spl.nactive
+
+    return NestedTransition(draw, logL, log_wt)
```

Comment:
Each `step!`, you should be extracting whatever the values are inside `NestedTransition` and putting them inside `spl.state.vi`, and then returning `vi(spl)`.